snapshots/snapshots-processor/src/main/kotlin/com/emergetools/snapshots/processor/utils/PreviewFinderUtils.kt

```diff
@@ -36,9 +36,44 @@ fun List<KSAnnotated>.functionsWithMultiPreviewAnnotation(
         .map { function ->
           function to getUniqueSnapshotConfigsFromMultiPreviewAnnotation(
             annotations = multiPreviewAnnotationPreviewAnnotations,
-            previewFunction = function
+            previewFunction = function,
           )
         }
     }
-    .toMap()
+
+  // The same function declaration can show up multiple times, so ensure the values are merged together
+  val mergedConfigs = mutableMapOf<KSFunctionDeclaration, List<ComposePreviewSnapshotConfig>>()
+  uniqueSnapshotConfigs.forEach {
+    mergedConfigs.putOrAppend(it.first, it.second)
+  }
+
+  return mergedConfigs
+}
+
+fun Resolver.getMultiPreviewAnnotations(): List<KSClassDeclaration> {
+  // Find all symbols with annotations and map to the annotation class declarations
+  val annotationClassDecls = getAllFiles()
+    .flatMap { it.declarations }
+    .filter { it.annotations.count() > 0 }
+    .flatMap { symbol ->
+      symbol.annotations.mapNotNull { annotation ->
+        val annotationQN = annotation.annotationType.resolve().declaration.qualifiedName
+        annotationQN?.let { qualifiedName ->
+          getClassDeclarationByName(qualifiedName)
+        }
+      }
+    }
+    .filter { it.classKind == ClassKind.ANNOTATION_CLASS }
```

Comment:
My only potential nit here would be to move this up above `.filter { it.annotations.count() > 0 }`, that way we're only working with annotation class declarations at that point, and don't risk running `annotation.count()` and `getClassDeclarationByName` on classes that are not annotation classes.

Reply:
Don't think that's possible since `it` is a `KSDeclaration` for the sequence of symbols with annotations, we haven't mapped to the annotation class declarations yet. 

I think the filter is actually redundant since we are flatmap'ing to the annotation class declaration, so it should only be a sequence of annotation classes at this point, but it does make it a little more readable so I'll leave it in.